modules/articles.py

```diff
@@ -0,0 +1,60 @@
+import nltk
+import newspaper
+import re
+
+class article_content():
+
+    def __init__(self, article_object):
+        #Whenever a class object is made, the article is tokenized and the dictionary file is opened as a set.
+        self.tokenized_words = nltk.word_tokenize(article_object.text)
+        self.article_object = article_object
+        with open('words_alpha.txt') as word_file:
+            self.valid_words = set(word_file.read().split())
+        
+        self.misspells = 0  # Set initial misspells to 0.
+
+    def citation_check(self):
+        
+        #Something NLTK stop words. These often aren't real citations. / to remove local files, and ad removal from scraped link.
+        self.not_citations = ("facebook.com", "twitter.com", "#", "mailto", "plus.google", "advertising", "advertisement", "ad.", "/", "whatsapp", "quora", "nav.", "instagram")
+
+        self.article_html = self.article_object.html
+        http = [self.article_object.url] + re.findall(r'href="[^\"]+', self.article_html) #Regex code for finding links, eg href="http://www.bbc.com" returns http://www.bbc.com.
```

Comment:
In line 21, we store the HTML text of the article in self.article_html.
Links in HTML are given like
`<a href="mylinkhere"`

So using regular expressions(Regex), we find out all the patterns that fit into the format `href="[^\"]+`

To explain this format, I will have to explain a little about regex. 'href="' means that whatever we are finding should contain that. So suppose we have `<a href="mylinkhere"`, the part `href=` from it will be found (for now).

Later, [^\"] means any character BUT ", because " will end the href tag. The + is to show that the last condition [^\"] i.e. any character BUT " is allowed will be repeated until a " is found.
Actually, for regex, you only need [^\] instead of [^\"],  but it's a " so need to escape it in Python when using "" "" brackets.

Anyway, seeing is believing, so you can check that it has extracted the URLS - 
![image](https://user-images.githubusercontent.com/22197137/64478143-86161c00-d1c1-11e9-8951-c6736b8683f2.png)

Reply:
You can see it in action here - 
https://regex101.com/r/Jeg4Qd/1